### ISSUE TITLE
Add on-screen keyboard for desktop use.

### ISSUE THREAD

**Issue by M0E-lnx** [novnc/noVNC#1632]
**Is your feature request related to a problem? Please describe.**
Not as much of an issue as much of an annoyance.  I have noticed when connecting from a mobile device, the noVNC toolbar presents a button to show the keyboard, but when connecting from a desktop (windows) it does not.

**Describe the solution you'd like**
Make the button visible so that the on-screen-keyboard can be launched from the noVNC toolbar

**Describe alternatives you've considered**
Physical hardware keyboard, but this is not always an option.  For example, when connecting from a MS Surface device, the system has an OSK, but I see no option to launch it from the noVNC toolbar.  As a workaround, i'm using osk apps on the server over vnc, which is not ideal depending on the desktop being served.

**Additional context**
The guacamole client offers similar functionality and it offers the OSK feature i'm looking for on noVNC

**Comment by CendioOssman** [comment]
That button is not exclusive to mobile devices, but rather to touch devices. So it would seem your browser isn't advertising touch support.

Can you give more details about your setup?

**Comment by M0E-lnx** [comment]
Firefox browser running on Win 11 from a MS Surface Pro 7 device connecting to Debian server.
I stand half corrected.  I do see the keyboard button on the toolbar, but clicking it does nothing.

**Comment by samhed** [comment]
Yeah, unfortunately, that's a known issue with the keyboard button in Firefox on Windows touch devices. I believe Edge and Chrome works.

**Comment by CendioOssman** [comment]
@M0E-lnx, could you confirm that this only happens with Firefox on you system as well?

And just to be clear, Guacamole works fine with Firefox for you?

**Comment by M0E-lnx** [comment]
Your are correct @CendioOssman, This seems to be limited to Firefox.  I dont have chrome, but I tried Edge (yuck!) and the OSK pops when I press the button there.

Also confirmed Guacamole works fine on Firefox

**Comment by CendioOssman** [comment]
Thanks for testing. The next step is putting together some test case so we can report this to Firefox.

**Comment by CendioOssman** [comment]
Hmm... I just tested this on Windows 10 and Firefox 94 and Firefox 99. Works just fine.

Which version of Firefox are you using?

And I assume Firefox pops up the OSK when using the address bar or input fields?

**Comment by M0E-lnx** [comment]
Using Firefox 99.0.1
FF does pop the OSK on input fields, but for whatever reason, not when I click the keyboard button on the NoVNC side toolbar (the vertical toolbar)

**Comment by CendioOssman** [comment]
Could you check the browser console if there are any warnings or errors there that might give a clue?

**Comment by M0E-lnx** [comment]
I'm not seeing any errors on the browser console, but I think I found a clue.

This is a surface tablet, but I do have a wireless keyboard setup for when I'm at home.  The dongle for the keyboard is so small that I normally just leave it plugged in at all times.  I'm guessing Firefox is detecting that hardware keyboard and refusing to pop the OSK.  Unplugging the dongle proves this theory.  With the dongle unplugged, the toolbar does pop the OSK.

I guess Edge and chrome must just ignore the hardware keyboard and pop it anyway.  I noticed on edge, if I pop the OSK, but start typing on the hardware keyboard, the OSK hides.  This functionality must be either not there or reversed in firefox.

**Comment by CendioOssman** [comment]
Thanks for the extra testing. So this still sounds like a Firefox issue rather than a noVNC one, so I'm doubtful we can do anything from our end.

Would you mind seeing if you can file a bug report with Firefox? They generally listen if Firefox behaves differently from the other browsers.

**Comment by sourcenux** [comment]
Hi,

when using Firefox 144.0/Vivaldi 7.6.3797.63/Chromium 141.0.6778.109, with Windows 10 (Client) and Proxmox 8.4.1 (Server) (NoVNC 1.6.0-2 and 1.6.0-3) I am also able to see the on-screen keyboard.
